apps/gsuite/mail_syncer/utils.py

```diff
@@ -0,0 +1,265 @@
+                     % (ow4_user.get_full_name(), ow4_user.id, ow4_user.online_mail))
+        return
+        # @ToDo: This should probably trigger an error or notification, so we easily can identify this issue.
+        # However, it should not stop execution of other, potentially safe, updates.
+
+    if not settings.OW4_GSUITE_SYNC.get('ENABLE_INSERT', False):
+        logger.debug('Skipping inserting user %s since ENABLE_INSERT is False.' % ow4_user)
+        return
+
+    group_key = get_group_key(domain, group_name)
+
+    g_suite_user_dict = {
+        'email': get_user_key(domain, ow4_user.online_mail),
+        'role': 'MEMBER',
+    }
+
+    directory = setup_g_suite_client()
+
+    logger.info("Inserting '%s' into G Suite group '%s'." % (ow4_user.online_mail, group_key))
+    resp = directory.members().insert(body=g_suite_user_dict, groupKey=group_key).execute()
+    logger.debug("Inserting response: %s" % resp)
+
+    return resp
+
+
+def remove_g_suite_user_from_group(domain, group_name, g_suite_user):
+    if isinstance(g_suite_user, str):
+        user_key = get_user_key(domain, g_suite_user)
+    else:
+        user_key = g_suite_user.get('email')
+    if 'leder@%s' % domain == user_key or 'nestleder@%s' % domain == user_key:
+        # Not removing these guys from any lists.
+        return
+
+    if not settings.OW4_GSUITE_SYNC.get('ENABLE_DELETE', False):
+        logger.debug('Skipping removing user %s since ENABLE_DELETE is False.' % g_suite_user)
+        return
+
+    group_key = get_group_key(domain, group_name)
+
+    directory = setup_g_suite_client()
+
+    logger.debug("Removing '%s' from G Suite group '%s'." % (user_key, group_key))
+    resp = directory.members().delete(groupKey=group_key, memberKey=user_key).execute()
+    logger.debug('Removal of user response: %s' % resp)
+
+    return resp
+
+
+def get_g_suite_users_for_group(domain, group_name):
+    # G Suite Group Key
+    group_key = get_group_key(domain, group_name)
+    logger.debug("Getting G Suite member list for '%s'." % group_key)
+
+    directory = setup_g_suite_client()
+
+    members = []
+    try:
+        members = directory.members().list(groupKey=group_key).execute().get('members')
+    except HttpError as e:
+        logger.error('HttpError when requesting user list: %s' % e)
+        # @ToDo: Find out how to handle this properly.
+        raise e
+
+    return members
+
+
+def get_g_suite_groups_for_user(domain, _user):
+    user = get_user(_user, gsuite=True)
+
+    user_key = get_user_key(domain, user)
+    logger.debug("Getting G Suite user membership list for '%s'." % user_key)
+
+    directory = setup_g_suite_client()
+
+    try:
+        groups = directory.groups().list(userKey=user_key).execute().get('groups')
+    except HttpError as e:
+        logger.error('HttpError when requesting user group membership list: %s' % e)
+        # @ToDo: Find out how to handle this properly.
```

Comment:
Not really. Since it might be triggered on a user save action it should probably not raise an exception and rather continue on, so the user is actually saved. I'll look into it.